recipes/libevent/all/conanfile.py

```diff
@@ -0,0 +1,113 @@
+from conans import ConanFile, CMake, tools
+import os
+import shutil
+
+
+class LibeventConan(ConanFile):
+    name = "libevent"
+    description = "libevent - an event notification library"
+    topics = ("conan", "libevent", "event")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/libevent/libevent"
+    author = "Bincrafters <bincrafters@gmail.com>"
+    license = "BSD-3-Clause"
+    exports_sources = ["CMakeLists.txt"]
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "with_openssl": [True, False],
+               "disable_threads": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "with_openssl": True,
+                       "disable_threads": False}
+    generators = "cmake"
+    _source_subfolder = "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.with_openssl and self.options.shared:
+            # static OpenSSL cannot be properly detected because libevent picks up system ssl first
+            # so enforce shared openssl
+            self.output.warn("Enforce shared OpenSSL for shared build")
+            self.options["openssl"].shared = self.options.shared
+
+    def requirements(self):
+        if self.options.with_openssl:
+            self.requires.add("openssl/1.1.1d")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = "libevent-{0}-stable".format(self.version)
+        if self.version == '2.1.11':
```

Comment:
I'm not sure if this workaround should be used for a generic recipe

Reply:
I agree with @uilianries, I think this is what the different version folders are for

Reply:
Sure, I can move this generic recipe to specific (2.1.11). However, the difference with the next recipe will be only in one line.